edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/TwinHandler.cs

```diff
@@ -0,0 +1,354 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.RegularExpressions;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Device;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class TwinHandler : ITwinHandler, IMessageConsumer, IMessageProducer
+    {
+        const string TwinGetDevice = "$edgehub/+/twin/get/#";
+        const string TwinGetModule = "$edgehub/+/+/twin/get/#";
+        const string TwinUpdateDevice = "$edgehub/+/twin/reported/#";
+        const string TwinUpdateModule = "$edgehub/+/+/twin/reported/#";
+
+        const string TwinGetPublishPattern = @"^\$edgehub/(?<id1>[^/\+\#]+)(/(?<id2>[^/\+\#]+))?/twin/get/\?\$rid=(?<rid>.+)";
```

Comment:
Nit: can we call it deviceId and moduleId (instead of id1 and id2). Same elsewhere.